src/algorithms/SortingAlgorithms.java

```diff
@@ -3,45 +3,122 @@
 public class SortingAlgorithms {
 
     public static void main(String[] args) {
-        int [] array = new int[]{2,1,8,3,5,5,-26};
-        sortByBubbleSorting(array);
+        int[] array = new int[]{2, 1, 8, 3, 5, 8, 444, 65, 5, -26};
+//        sortByBubbleSorting(array);
+//        sortByInsertionSorting(array);
+//        sortBySelectionSorting(array);
+//        sortByMergeSorting(array);
+        sortByQuickSorting(array);
         print(array);
     }
 
     public static void sortByBubbleSorting(int[] array) {
         for (int i = 0; i < array.length; i++) {
-            for (int j = 0; j < array.length-i-1; j++) {
-                if (array[j] > array[j+1]) {
-                    swap(array, j+1, j);
+            for (int j = 0; j < array.length - i - 1; j++) {
+                if (array[j] > array[j + 1]) {
+                    swap(array, j + 1, j);
                 }
             }
         }
     }
 
-    public static void sortBySelectionSorting(int[] array){
+    public static void sortBySelectionSorting(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            int minIndex = i;
+            for (int j = i; j < array.length; j++) {
+                if (array[j] < array[minIndex]) {
+                    minIndex = j;
+                    swap(array, minIndex, i);
+                }
+            }
+        }
+    }
+
```

Comment:
it doesn't work correctly. Please try again.
https://en.wikipedia.org/wiki/Insertion_sort

Reply:
Correcred.

Reply:
Corrected.